src/App.tsx

```diff
@@ -1,26 +1,58 @@
-import React from 'react';
-import logo from './logo.svg';
+import React from "react";
 import styles from './App.module.scss';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Link,
+  useRouteMatch,
+  useParams
+} from "react-router-dom";
+import {Home} from "./Homepage/Homepage";
+import {RoverExplorer} from "./RoverExplorer/RoverExplorer";
+import {TimelineID} from "./TimelineID/TimelineID";
 
-function App() {
+
+export default function App() {
   return (
-    <div className={styles.App}>
-      <header className={styles.appHeader}>
-        <img src={logo} className={styles.appLogo} alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className={styles.appLink}
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <div className={styles.navbar}>
+        <ul>
+          <li>
+            <Link to="/">Home</Link>
+          </li>
+          <li>
+            <Link to="/rover_explorer">Rover explorer</Link>
+          </li>
+          <li>
+            <Link to="/timeline/opportunity">Opportunity</Link>
+          </li>
+          <li>
+            <Link to="/timeline/spirit">Spirit</Link>
+          </li>
+          <li>
+            <Link to="/timeline/curiosity">Curiosity</Link>
+          </li>
+          <li>
+            <Link to="/timeline/perserverance">Perserverance</Link>
+          </li>
+        </ul>
+
+        <Switch>
+          <Route path="/rover_explorer">
+            <RoverExplorer />
+          </Route>
+          <Route path="/timeline/:id">
+           {<TimelineID />}
+            </Route> 
+          <Route path="/">
+            <Home />
+          </Route>
+        </Switch>
```

Comment:
shouldn't this be outside of the NavBar?
(e - isn't this the page content not the navbar content?